external/js/cothority/lib/skipchain/skipchain.js

```diff
@@ -134,83 +136,107 @@ class Client {
 
     // get the roster aggregate key and subtract any exception listed.
     const aggregate = roster.aggregateKey();
-
-    // all indices of the absent nodes from the roster
-    const absenteesIdx = getSetBits(bitmask);
-    // compute reduced public key
-    absenteesIdx.forEach(idx => {
-      aggregate.sub(aggregate, roster.get(idx));
-    });
-
-    // commitment to subtract from the signature
-    const absentCommitment = this.group.point().null();
-    if (link.exceptions) {
-      const excLength = link.exceptions.length;
-      for (var i = 0; i < excLength; i++) {
-        var exception = link.exceptions[i];
-        // subtract the absent public key from the roster aggregate key
-        aggregate.sub(aggregate, roster.get(exception.index));
-        // aggregate all the absent commitment
-        var individual = group.point();
-        individual.unmarshalBinary(exception.commitment);
-        absentCommitment.add(absentCommitment, individual);
-      }
-    }
+    //compute reduced public key
+    /*absenteesIdx.forEach(idx => {*/
+    //aggregate.sub(aggregate, roster.get(idx));
+    /*});*/
 
     // XXX suppose c = H(R || Pub || m) , with R being the FULL commitment
     // that is being generated at challenge time and the signature is
     // R' || s with R' being the reduced commitment
     // R' = R - SUM(exception-commitment)
-    const R = group.point();
-    R.unmarshalBinary(link.signature.slice(0, pointLen));
-    const reducedR = R.clone();
-    reducedR.sub(reducedR, commitment);
-    const s = group.scalar();
-    s.unmarshalBinary(link.signature.slice(pointLen, pointLen + scalarLen));
-
+    const R = this.group.point();
+    R.unmarshalBinary(bftSig.signature.slice(0, pointLen));
+    const s = this.group.scalar();
+    s.unmarshalBinary(bftSig.signature.slice(pointLen, pointLen + scalarLen));
     // recompute challenge = H(R || P || M)
     // with P being the roster aggregate public key minus the public keys
     // indicated by the bitmask
-    const buffPub = publicKey.marshalBinary();
-    const challenge = schnorr.hashSchnorr(
-      suite,
+    const challenge = hashSchnorr(
+      this.group,
       R.marshalBinary(),
       aggregate.marshalBinary(),
       message
     );
+    // compute -(c * Aggregate)
+    const mca = this.group.point().neg(aggregate);
+    mca.mul(challenge, mca);
     // compute sG
-    const left = suite.point().mul(s, null);
+    const left = this.group.point().mul(s, null);
+    left.add(left, mca);
     // compute R + challenge * Public
-    const right = suite.point().mul(challenge, publicKey);
-    right.add(right, reducedR);
+    //const right = this.group.point().mul(challenge, aggregate);
+    //right.add(right, R);
+    const right = R;
     if (!right.equal(left)) {
-      return new Error("invalid signature");
+      //return new Error("invalid signature");
+      console.log("invalid signature...");
+      return false;
     }
     return null;
```

Comment:
The documentation suggests the return type is boolean but we're returning an Error/false on failure and null on success.